File: android/src/app/organicmaps/compat/CompatV23.java
package app.organicmaps.compat;

import android.annotation.TargetApi;
import android.util.Log;
import android.widget.TimePicker;

@TargetApi(23)
public class CompatV23 extends CompatV21 implements Compat {

  @Override
  public void setTime(TimePicker picker, int hour, int minute) {
    picker.setHour(hour);
    picker.setMinute(minute);
  }

  @Override
  public int getHour(TimePicker picker) {
    return picker.getHour();
  }

  @Override
  public int getMinute(TimePicker picker) {
    return picker.getMinute();
  }
}

File: android/src/app/organicmaps/compat/CompatV23.kt
package app.organicmaps.compat

import android.annotation.TargetApi
import android.widget.TimePicker

@TargetApi(23)
open class CompatV23: CompatV21(), Compat {

  override fun setTime(picker: TimePicker, hour: Int, minute: Int) {
    picker.hour = hour
    picker.minute = minute
  }

  override fun getHour(picker: TimePicker): Int {
    return picker.hour
  }

  override fun getMinute(picker: TimePicker): Int {
    return picker.minute
  }
}